v-next/hardhat-utils/src/global-dir.ts

```diff
@@ -0,0 +1,51 @@
+import { generatePaths, HARDHAT_PACKAGE_NAME } from "./internal/global-dir.js";
+
+import { ensureDir } from "@ignored/hardhat-vnext-utils/fs";
+
+/**
+ * Returns the configuration directory path for a given package (defaults to "hardhat").
+ * Ensures that the directory exists before returning the path.
+ *
+ * @param packageName The name of the package for which to generate paths. Defaults to "hardhat" if no package name is provided.
+ * @returns The path to the hardhat configuration directory.
+ * @throws FileSystemAccessError for any error.
+ */
+export async function getConfigDir(
+  packageName: string = HARDHAT_PACKAGE_NAME,
```

Comment:
I added the possibility to pass a `package name` in every exported function so that user can override the default behavior for all of them